docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,232 @@
-in Dask-ML. The appropriate one to use depends on the size of your dataset and
-whether the underlying estimator implements the `partial_fit` method.
+Hyperparameter searches are a required process in machine learning. Briefly,
+these searches require finding optimal values for a model that can not be found
+with the data. For more detail, see "`Tuning the hyper-parameters of an
+estimator <https://scikit-learn.org/stable/modules/grid_search.html>`_."
 
-If your dataset is relatively small or the underlying estimator doesn't implement
-``partial_fit``, you can use :class:`dask_ml.model_selection.GridSearchCV` or
-:class:`dask_ml.model_selection.RandomizedSearchCV`.
-These are drop-in replacements for their scikit-learn counterparts, that should offer better performance and handling of Dask Arrays and DataFrames.
-The underlying estimator will need to be able to train on each cross-validation split of the data.
-See :ref:`hyperparameter.drop-in` for more.
+These searches often take weeks or months to complete, typically when searching
+for great performance or with massive datasets. This is common when preparing
+for production or a paper publication. These issues will be clarified in the
+following section:
 
-If your data is large and the underlying estimator implements ``partial_fit``, you can
-Dask-ML's :ref:`*incremental* hyperparameter optimizers <hyperparameter.incremental>`.
+* ":ref:`hyperparameter.scaling`" will mention problems often seen in
+  hyperparameter optimization searches, and tools to get around these issues.
+
+Then, some tooling will be mentioned to circumvent the issues mentioned, and
+expanded on in the following sections:
+
+1. ":ref:`hyperparameter.drop-in`" will mention classes that are drop-in
+   replacements for Scikit-learn but work nicely with Dask objects.
+2. ":ref:`hyperparameter.incremental`" will mention classes that work well with
+   large datasets.
+3. ":ref:`hyperparameter.adaptive`" will mention classes that avoid extra
+   computation and find high-performing hyperparameters more quickly.
+
+.. _hyperparameter.scaling:
+
+Scaling hyperparameter searches
+-------------------------------
+
+Dask-ML provides classes to avoid the two most common issues in hyperparameter
+optimization, when the hyperparameter search is...
+
+1. "**memory constrained"**. This happens when the dataset size is too large to
+   fit in memory.  This typically happens when a model needs to be tuned for a
+   larger-than-memory dataset after local development.
+2. "**compute constrained**". This happen when the computation takes too long
+   even with data that can fit in memory.  This typically happens when many
+   hyperparameters need to be tuned, which especially common with neural
+   networks.
+
+"Memory constrained" searches happen when the data doesn't fit in the memory of
+a single machine:
+
+.. code:: python
+
+   # not memory constrained
+   import pandas as pd
```

Comment:
Format these like you're at the interpreter. Start with `>>>` and put the outputs below.